### ISSUE TITLE
Building Debian package requires internet access

### ISSUE THREAD

**Issue by SwooshyCueb** [mono/linux-packaging-msbuild#1]
Even with all build-dependencies satisfied, building the msbuild package still requires internet access. This goes against Debian's (Upstream Guide)[https://wiki.debian.org/UpstreamGuide#No_Downloads] and makes building this package (impossible on launchpad)[https://launchpadlibrarian.net/333239305/buildlog_ubuntu-xenial-amd64.msbuild_15.2+xamarinxplat.2017.05.04.00.44-0xamarin3+ubuntu1404b1+ors0_BUILDING.txt.gz].

**Comment by DragonSA** [comment]
I have the same problem when building for FreeBSD.  I believe msbuild downloads the following - during build:
 1. **dotnet sdk**: this can be pre-downloaded and extracted into the appropriate location (see `init-tools.sh`)
 2. **build tools**: I believe the same approach as above can be used, I took a different approach on FreeBSD.
 3. **many packages**: these can be pre-downloaded and extracted into `$HOME/.nuget/packages`.  This is a bit involved as a particular structure (not in the archive) needs to be followed - and there isn't an easy was to get the full list of packages required - and most packages are not required to build.

**Comment by ArsenShnurkov** [comment]
The same on Gentoo. I am trying to split this into three packages:
- one for Microsoft.Build.Framework.dll and Microsoft.Build.Utils.Core.dll
- one for Microsoft.DotNet.Build.Tasks.dll (it is necessary to compile resources)
- and the last for msbuild.exe, Microsoft.Build.dll and Microsoft.Build.Tasks.Core.dll
First package is necessary, because it exposes API for writing tasks, and will be used as dependency for many other packages, second one is necessary for building Microsoft.Build.Tasks.Core.dll and resides in separate repository.

**Comment by EoflaOE** [comment]
Yes. I have the same problem as the build process for my application, Kernel Simulator, requires at least MSBuild 15. There was no "msbuild" package for Debian d it's derivatives. I have uploaded KS to Launchpad, and the build failed for the exact same reason (tries to get files off Internet)

I am thinking into forking this repo and modifying it so it works without Internet and use that to build MSBuild without Internet access, so that it works on Launchpad. Is that acceptable?

**Comment by ArsenShnurkov** [comment]
There is one more problem with new project format. I am trying to package Microsoft.NET.Sdk, and it depend on some nuget parts, because new .csproj-files contain <PackageReference ... /> items. The functions for working with packages in Sdk should be reimplemented with pkg-config instead of NuGet.

**Comment by EoflaOE** [comment]
I have managed to get MSBuild to build (debuild -sa) using local packages, local .NET Core .tar.gz file, and some modified files so they work **offline** without errors. However, I have to separate large packages directories from the source code so that I don't exceed the repo size too fast. I will upload the packages .tar.gz on a separate secure place. Afterwards, I will make the instructions on how to prepare the source code for building and uploading to Debian FTP, Launchpad, etc. After they are all finished, the changes will be put to my fork.

**Comment by EoflaOE** [comment]
Well, all of the packages are tarred, gzipped, and released to the Releases of my fork. Now, it's ready to use.

Repository link: https://github.com/EoflaOE/linux-remote-packaging-msbuild
The instructions are ready at https://github.com/EoflaOE/linux-remote-packaging-msbuild/blob/master/documentation/building-msbuild-offline.md

This will be the first step of deployability to Launchpad or alternatives.
